sup3r/pipeline/forward_pass.py

```diff
@@ -918,17 +996,30 @@ def _run_parallel(self, data, max_workers=None):
         """
         futures = {}
         now = dt.now()
+        logger.info('Starting process pool with {} workers'
+                    .format(max_workers))
         with SpawnProcessPool(max_workers=max_workers) as exe:
-            for i, (sh, slp, shc) in enumerate(zip(
-                    self.hr_slices, self.lr_pad_slices,
-                    self.hr_crop_slices)):
-                future = exe.submit(
-                    ForwardPass.forward_pass_chunk,
-                    data_chunk=self.data_handler.data[slp],
-                    crop_slices=shc, model_path=self.model_path)
+            zip_iter = zip(self.hr_slices, self.lr_pad_slices,
+                           self.hr_crop_slices)
+            for i, (sh, slp, shc) in enumerate(zip_iter):
+
+                future = exe.submit(ForwardPass.forward_pass_chunk,
```

Comment:
I see that you only include the option to pass a loaded model for the serial routine. I experimented with doing this for the parallel routine and got layer errors about mismatched sizes with cached data. I'm not sure the serial routine will avoid this problem either?

Reply:
I havent been running on cached data but yeah my thought right now is pass the model in for serial and load the model in parallel workers. It's worked for me but I havent been running with cached data. Is there a test for that?